### PR TITLE
Add beforePing callback + rename beforeServerInfo option

### PR DESCRIPTION
**Pull request by RichardDorian**
Hello it's me again 👋

I fixed a little issue with the `server_info`/`ping` packet. Before the `beforePing` options was controlling the `server_info` packet and there was no control over the `ping` packet. I renamed the `beforePing` function to `beforeServerInfo` **(So yes there's a breaking change!)** and i "added" an option `beforePing` that will override the answerToPing function, that i created as well, if exists.

For the breaking change:
```js
// Before
const serverOptions = {
    ...serverOptions
    beforePing: async (response, client, callback) => {
      response = await ping({
        host: config.server.host,
        port: config.server.port,
        version: client.version,
      });
      hypixelLatency = response.latency;
      response.players.sample = [{ name: '§cSolar Stats Proxy', id: NIL }];

      callback(null, response);
    },
  },
```
```js
// After
const serverOptions = {
    ...serverOptions
    beforeServerInfo: async (response, client, callback) => {
      response = await ping({
        host: config.server.host,
        port: config.server.port,
        version: client.version,
      });
      hypixelLatency = response.latency;
      response.players.sample = [{ name: '§cSolar Stats Proxy', id: NIL }];

      callback(null, response);
    },
  },
```

And now you have control over the ping packet. So for example (the main use I think) can delay the packet to spoof the ping value displayed by the client

```js
// Bofore
// Can't do anything
```

```js
// After
const serverOptions = {
    ...serverOptions
    beforePing: async (client, packet) => {
      setTimeout(() => {
        client.write('ping', { time: packet.time });
        client.end();
      }, 1500); // 1.5k ping wow!
    },
  },
```

image

Yeah most of the code is extracted from my proxy because uh i was a bit lazy to write fresh new one. And yeah I have to admit, I did this because i need to spoof the ping value but i guess someone else will maybe need this in the future so

### CODE REVIEW

**Review by andriycraft — APPROVED** [review]

**Comment by rom1504** [comment]
hi, can you fix the conflicts?

**Comment by RichardDorian** [comment]
Hopefully i fixed them all

**Comment by MauritsWilke** [comment]
+1 Please consider this again :)

File: docs/API.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ automatically logged in and validated against mojang's auth.
  * kickTimeout : default to `10*1000` (10s), kick client that doesn't answer to keepalive after that time
  * checkTimeoutInterval : default to `4*1000` (4s), send keepalive packet at that period
  * online-mode : default to true
- * beforePing : allow customisation of the answer to ping the server does. 
+ * beforeServerInfo : allow customisation of the answer to server info (MOTD, server icon, player list and so on) the server does.
```
Comment (rom1504) [inline docs/API.md:15]: can you make it not breaking

**Comment by rom1504** [comment]
this is breaking and use case is not clear

**Pull request closed without merging**